gamemodes/terrortown/gamemode/server/sv_main.lua

```diff
@@ -1191,6 +1196,17 @@ function BeginRound()
 
 	ARMOR:InitPlayerArmor()
 
+	local plys = player.GetAll()
+
+	for i = 1, #plys do
+		local ply = plys[i]
+
+		ply:SetDiedInRound(false)
+
+		-- a player should be considered "was active in round" if they received a role
+		ply:SetActiveInRound(ply:Alive() and ply:IsTerror())
```

Comment:
Is it really needed? There are functions like `ply:GetForceSpec()` i think

Reply:
Maybe there is something that does the same already. I wasn't able to find anything and nobody in the dev group had an idea when I asked
Forced spec is something else though, latejoiners aren't forced spectators

Reply:
LateJoiners should be forced specs as well, if not, it's a bug

Reply:
aren't forced specs players that have set the "force spec" checkbox?

Reply:
If so, there are bugs based by the idea that forced specs are players that were forced into the spec team (LateJoiners or CheckBox(by own or automatically if being idle))

Reply:
Idle autosets the checkbox though, latejoining doesn't.
So there has to be a difference somewhere.

Reply:
Then it's something we need to keep in mind here: https://github.com/TTT-2/TTT2/pull/517/files#diff-1512bbb3506493de161ace83dfe6b206R41

Reply:
Okay, I checked this code. Why should I change it to `ply:GetForceSpec()`? In my opinion my implementation is correct because a players has to be alive and not in the spectator team to be counted as "being active at round start". Forced spectator only sets the networked bool which might not cover all cases.

However `ply:SetActiveInRound(ply:IsTerror())` could be sufficient.

Reply:
Maybe you missed the entire discussion and the last meeting or the break was a bit too long 😅 You shouldn‘t replace it by GetForceSpec, but check whether code snippets that used GetForceSpec before need to change into your new system

Reply:
But my system has nothing to do with spectators / forced specators